File: SAMSUI/src/app/Model/base/tax.ts
export class taxModel{

     public taxId ?: number;

     public orgId ?: number;

     public locationId ?: number;  
     
     public locationName ?: string;

     public taxCode ?: string;

     public tacRate ?: number; 

     public taxDesc ?: string; 
     
     public active ?: boolean; 

     public createdBy ?: string;
     
     public createdDt ?: Date;  
     
     public updatedBy ?: string; 
     
     public updatedDt ?: Date;   

     public select ?: boolean;

     public taxComputation ?: object;

     public taxRate ?: string;

     public remarks ?: string;

     public pageNumber ?: Number;

     public recordsPerPage ?: Number;

     public direction ?: string;

     public columnName ?: string;

}

File: sams-backend/src/main/java/com/sams/entity/Tax.java
package com.sams.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "m_tax", schema = "base")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Tax {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "tax_id")
    private Long taxId;

    @Column(name = "org_id")
    private Long orgId;

    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "location_name")
    private String locationName;

    @Column(name = "tax_code")
    private String taxCode;

    @Column(name = "tax_rate")
    private Double taxRate;

    @Column(name = "tax_desc")
    private String taxDesc;

    @Column(name = "tax_computation")
    private String taxComputation;

    @Column(name = "active")
    private Boolean active;

    @Column(name = "remarks")
    private String remarks;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "created_dt")
    private LocalDateTime createdDt;

    @Column(name = "updated_by")
    private String updatedBy;

    @Column(name = "updated_dt")
    private LocalDateTime updatedDt;

}